Added ability to localize based on user's location

azilnik opened the pull request:
Instead of listing position options as elsewhere, it might be possible to localize to the user's own location, using the user's IP to detect where he or she is, and render in the site copy.

This would help make possible remote workers feel encouraged to apply.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/join-dev-design?pullRequest=161) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/join-dev-design?pullRequest=161) before we can accept your contribution.<br/><br/>:x: azilnik [sign now](https://cla.opensource.microsoft.com/Microsoft/join-dev-design?pullRequest=161)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/join-dev-design?pullRequest=161) it.</sub>

leovarmak commented:
This is nice! 
Did you forget to remove the API Key ?

azilnik commented:
@leovarmak good call out! I removed the API Key. @prichodko, I added a comment in `location.js` about including an API key if this were to be merged. Here is what it looks like with the API key all set up properly.

<img width="636" alt="screen shot 2018-10-09 at 5 36 11 pm" src="https://user-images.githubusercontent.com/1574589/46700200-20549500-cbea-11e8-9cd4-887b91411cc0.png">

prichodko commented:
Hey @azilnik, thanks for you contribution! Although it's a neat feature I am afraid it could confuse people into thinking there is a Microsoft office hiring in their city, which probably is not true. What do you think

azilnik commented:
It's a good call-out @prichodko! I think it would work if there are remote opportunities. Hmm I wonder if there's a way to get the user's location data and match it to open roles, and indicate whether you're hiring in that particular location. Thoughts?
